Add Alexa.ChannelController api function. Add StepSpeaker volume step functionality

Dilbert66 opened the pull request:
Added missing Alexa.ChannelController functions.  Specifically ChangeChannel and SkipChannel commands.  These functions will call the play_media function in a media_player app if it has the capability published and pass on the channel# or channel name.  The selected media player can then use this to select the channel on the device it is associated to.
Modified the existing Alexa.StepSpeaker Setvolume function to actually do a stepped volume change using the steps sent by Alexa. The Alexa default step of 10 for a simple volume up/down can be changed via an exposed media_player attribute called volume_step_default.  A better value would be 5.  Any other value then default will be sent as sequential volume  up /down to the media_player.

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

No extra configuration needed for the Alexa component.  These changes has been tested on a live system.

Dilbert66 committed:
Added channel controller functions
Modified StepSpeaker volume function to actually do up/down steps

Dilbert66 committed:
Added channel controller functions
Modified StepSpeaker volume function to actually do up/down steps

Dilbert66 committed:
Correct typo in attribute

Dilbert66 committed:
Correct typo in attribute

Dilbert66 committed:
changed attribute lookup for StepSpeaker to use device_state_attributes

Dilbert66 committed:
changed attribute lookup for StepSpeaker to use device_state_attributes

shred86 committed:
Add abode config entries and device registry (#26699)

* Adds support for config entries and device registry

* Fixing string formatting for logger

* Add unit test for abode config flow

* Fix for lights, only allow one config, add ability to unload entry

* Fix for subscribing to hass_events on adding abode component

* Several fixes from code review

* Several fixes from second code review

* Several fixes from third code review

* Update documentation url to fix branch conflict

* Fixes config flow and removes unused constants

* Fix for switches, polling entry option, improved tests

* Update .coveragerc, disable pylint W0611, remove polling from UI

* Multiple fixes and edits to adhere to style guidelines

* Removed unique_id

* Minor correction for formatting error in rebase

* Resolves issue causing CI to fail

* Bump abodepy version

* Add remove device callback and minor clean up

* Fix incorrect method name

* Docstring edits

* Fix duplicate import issues from rebase

* Add logout_listener attribute to AbodeSystem

* Add additional test for complete coverage

Dilbert66 committed:
Lint fixes

Dilbert66 committed:
Lint fixes

starkillerOG committed:
Allow MQTT json light floating point transition (#27253)

* MQTT json light: allow floating point transition

Allow to use floating point values for the transition time of the MQTT json light.
In this way transitions shorter than 1s can be used (0.5 seconds for instance) if the MQTT light supports it.

* Always sent a float

Dilbert66 committed:
Lint fixes

Dilbert66 committed:
Lint fixes

Dilbert66 committed:
More lint fixes

Dilbert66 committed:
More lint fixes

ochlocracy commented on homeassistant/components/alexa/entities.py:
```diff
@@ -51,6 +51,7 @@
     AlexaStepSpeaker,
     AlexaTemperatureSensor,
     AlexaThermostatController,
+    AlexaChannelController
```
Sort imports alphabetically.

ochlocracy commented on homeassistant/components/alexa/capabilities.py:
```diff
@@ -716,3 +716,10 @@ def configuration(self):
+    """Implements Alexa.ChannelController. https://developer.amazon.com/docs/device-apis/alexa-channelcontroller.html """
+
+    def name(self):
+        return 'Alexa.ChannelController'
```
```suggestion
        return 'Alexa.ChannelController'
     
```
Need newline at end of file.

ochlocracy commented on homeassistant/components/alexa/capabilities.py:
```diff
@@ -716,3 +716,10 @@ def configuration(self):
+
+
+class AlexaChannelController(AlexaCapibility):
+    """Implements Alexa.ChannelController. https://developer.amazon.com/docs/device-apis/alexa-channelcontroller.html """
```
```suggestion
    """Implements Alexa.ChannelController. 
    
    https://developer.amazon.com/docs/device-apis/alexa-channelcontroller.html
    """
```
Match formatting of other Controllers.

Dilbert66 committed:
Lint, formatting fixes

Dilbert66 committed:
Lint, formatting fixes

Dilbert66 committed:
Lint fixes

Dilbert66 committed:
Lint fixes

ochlocracy commented on homeassistant/components/alexa/handlers.py:
```diff
@@ -924,3 +943,53 @@ def temperature_from_object(hass, temp_obj, interval=False):
+        channel = directive.payload['channel']['callSign']
+    elif 'name' in directive.payload['channelMetadata'].keys():
+        channel = directive.payload['channelMetadata']['name']
+
```
According the Alexa API `channel.number`, `channelMetadata.name`, `channel.callSign`, `channel.affiliateCallSign` or `channel.uri` will be provided in the directive. Is it possible to handle the `channel.affiliateCallSign` and `channel.uri` too?

ochlocracy commented on homeassistant/components/alexa/handlers.py:
```diff
@@ -924,3 +943,53 @@ def temperature_from_object(hass, temp_obj, interval=False):
+            entity.domain, SERVICE_MEDIA_PREVIOUS_TRACK,
+            data, blocking=False, context=context)
+
+    return directive.response()
```
```suggestion
    response = directive.response()
    response.add_context_property(
        {
            "namespace": "Alexa.ChannelController",
            "name": "channel",
            "value": {
              "number": "1234",
              "callSign": "callsign1",
              "affiliateCallSign": "callsign2"
            }
        )
        
        return response
```
According the the Alexa API:

> You must send an Response event if a request to change or skip channels was successful. The response should include a context object that reports the channel after the request completes.

Dilbert66 commented:
Good idea. I will do that.  Still working on the formatting . You guys are
strict ! lol
Alain

On Sun, Oct 13, 2019 at 3:11 PM ochlocracy <notifications@github.com> wrote:

> *@ochlocracy* commented on this pull request.
> ------------------------------
>
> In homeassistant/components/alexa/handlers.py
> <https://github.com/home-assistant/home-assistant/pull/27605#discussion_r334291752>
> :
>
> > +    if 'number' in directive.payload['channel'].keys():
> +        channel = directive.payload['channel']['number']
> +    elif 'callSign' in directive.payload['channel'].keys():
> +        channel = directive.payload['channel']['callSign']
> +    elif 'name' in directive.payload['channelMetadata'].keys():
> +        channel = directive.payload['channelMetadata']['name']
> +
>
> According the Alexa API channel.number, channelMetadata.name,
> channel.callSign, channel.affiliateCallSign or channel.uri will be
> provided in the directive. Is it possible to handle the
> channel.affiliateCallSign and channel.uri too?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/27605?email_source=notifications&email_token=ABW4E7LJDUEUZXNJVCQOAS3QONXGBA5CNFSM4JAH6NTKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCHY5O6Y#pullrequestreview-301062011>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABW4E7JQD6MWRMNGX655RYLQONXGBANCNFSM4JAH6NTA>
> .
>

Dilbert66 commented:
Yes, I saw that in the api docs. Worked without it but best to add the
context section. I will do that as well.

On Sun, Oct 13, 2019 at 3:23 PM ochlocracy <notifications@github.com> wrote:

> *@ochlocracy* commented on this pull request.
> ------------------------------
>
> In homeassistant/components/alexa/handlers.py
> <https://github.com/home-assistant/home-assistant/pull/27605#discussion_r334292228>
> :
>
> > +
>
> +    data = {
>
> +        ATTR_ENTITY_ID: entity.entity_id
>
> +    }
>
> +
>
> +    if channel > 0:
>
> +        await hass.services.async_call(
>
> +            entity.domain, SERVICE_MEDIA_NEXT_TRACK,
>
> +            data, blocking=False, context=context)
>
> +
>
> +    if channel < 0:
>
> +        await hass.services.async_call(
>
> +            entity.domain, SERVICE_MEDIA_PREVIOUS_TRACK,
>
> +            data, blocking=False, context=context)
>
> +
>
> +    return directive.response()
>
>
> ⬇️ Suggested change
>
> -    return directive.response()
>
> +    response = directive.response()
>
> +    response.add_context_property(
>
> +        {
>
> +            "namespace": "Alexa.ChannelController",
>
> +            "name": "channel",
>
> +            "value": {
>
> +              "number": "1234",
>
> +              "callSign": "callsign1",
>
> +              "affiliateCallSign": "callsign2"
>
> +            }
>
> +        )
>
> +
>
> +        return response
>
>
> According the the Alexa API:
>
> You must send an Response event if a request to change or skip channels
> was successful. The response should include a context object that reports
> the channel after the request completes.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/27605?email_source=notifications&email_token=ABW4E7MX7SMMK2CQCRQXH5DQONYRJA5CNFSM4JAH6NTKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCHY5SMY#pullrequestreview-301062451>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABW4E7IC2BOBD2HYGSCW343QONYRJANCNFSM4JAH6NTA>
> .
>

Dilbert66 commented:
The only issue is that channel state is not a given when using skipchannel
as there is no feedback on knowing the current channel and therefore the
returned context will most likely not have the actual tuned channel.  In
this case, a fixed value of 0000 or 9999 perhaps will be given.  With the
changechannel command , it's easy, just return the value received on the
original request. Will need to try and see the results.

On Sun, Oct 13, 2019 at 3:25 PM Alain Turbide <aturbide99@gmail.com> wrote:

> Yes, I saw that in the api docs. Worked without it but best to add the
> context section. I will do that as well.
>
> On Sun, Oct 13, 2019 at 3:23 PM ochlocracy <notifications@github.com>
> wrote:
>
>> *@ochlocracy* commented on this pull request.
>> ------------------------------
>>
>> In homeassistant/components/alexa/handlers.py
>> <https://github.com/home-assistant/home-assistant/pull/27605#discussion_r334292228>
>> :
>>
>> > +
>>
>> +    data = {
>>
>> +        ATTR_ENTITY_ID: entity.entity_id
>>
>> +    }
>>
>> +
>>
>> +    if channel > 0:
>>
>> +        await hass.services.async_call(
>>
>> +            entity.domain, SERVICE_MEDIA_NEXT_TRACK,
>>
>> +            data, blocking=False, context=context)
>>
>> +
>>
>> +    if channel < 0:
>>
>> +        await hass.services.async_call(
>>
>> +            entity.domain, SERVICE_MEDIA_PREVIOUS_TRACK,
>>
>> +            data, blocking=False, context=context)
>>
>> +
>>
>> +    return directive.response()
>>
>>
>> ⬇️ Suggested change
>>
>> -    return directive.response()
>>
>> +    response = directive.response()
>>
>> +    response.add_context_property(
>>
>> +        {
>>
>> +            "namespace": "Alexa.ChannelController",
>>
>> +            "name": "channel",
>>
>> +            "value": {
>>
>> +              "number": "1234",
>>
>> +              "callSign": "callsign1",
>>
>> +              "affiliateCallSign": "callsign2"
>>
>> +            }
>>
>> +        )
>>
>> +
>>
>> +        return response
>>
>>
>> According the the Alexa API:
>>
>> You must send an Response event if a request to change or skip channels
>> was successful. The response should include a context object that reports
>> the channel after the request completes.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/home-assistant/home-assistant/pull/27605?email_source=notifications&email_token=ABW4E7MX7SMMK2CQCRQXH5DQONYRJA5CNFSM4JAH6NTKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCHY5SMY#pullrequestreview-301062451>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ABW4E7IC2BOBD2HYGSCW343QONYRJANCNFSM4JAH6NTA>
>> .
>>
>

Dilbert66 commented:
What do you think of also implementing the step portion in the skip
channel.  For now, it treats it as 1.  Adding a for loop wil allow the user
to skip however many channels they require, just like the StepSpeaker
function.
Easy to do.

On Sun, Oct 13, 2019 at 3:34 PM Alain Turbide <aturbide99@gmail.com> wrote:

> The only issue is that channel state is not a given when using skipchannel
> as there is no feedback on knowing the current channel and therefore the
> returned context will most likely not have the actual tuned channel.  In
> this case, a fixed value of 0000 or 9999 perhaps will be given.  With the
> changechannel command , it's easy, just return the value received on the
> original request. Will need to try and see the results.
>
> On Sun, Oct 13, 2019 at 3:25 PM Alain Turbide <aturbide99@gmail.com>
> wrote:
>
>> Yes, I saw that in the api docs. Worked without it but best to add the
>> context section. I will do that as well.
>>
>> On Sun, Oct 13, 2019 at 3:23 PM ochlocracy <notifications@github.com>
>> wrote:
>>
>>> *@ochlocracy* commented on this pull request.
>>> ------------------------------
>>>
>>> In homeassistant/components/alexa/handlers.py
>>> <https://github.com/home-assistant/home-assistant/pull/27605#discussion_r334292228>
>>> :
>>>
>>> > +
>>>
>>> +    data = {
>>>
>>> +        ATTR_ENTITY_ID: entity.entity_id
>>>
>>> +    }
>>>
>>> +
>>>
>>> +    if channel > 0:
>>>
>>> +        await hass.services.async_call(
>>>
>>> +            entity.domain, SERVICE_MEDIA_NEXT_TRACK,
>>>
>>> +            data, blocking=False, context=context)
>>>
>>> +
>>>
>>> +    if channel < 0:
>>>
>>> +        await hass.services.async_call(
>>>
>>> +            entity.domain, SERVICE_MEDIA_PREVIOUS_TRACK,
>>>
>>> +            data, blocking=False, context=context)
>>>
>>> +
>>>
>>> +    return directive.response()
>>>
>>>
>>> ⬇️ Suggested change
>>>
>>> -    return directive.response()
>>>
>>> +    response = directive.response()
>>>
>>> +    response.add_context_property(
>>>
>>> +        {
>>>
>>> +            "namespace": "Alexa.ChannelController",
>>>
>>> +            "name": "channel",
>>>
>>> +            "value": {
>>>
>>> +              "number": "1234",
>>>
>>> +              "callSign": "callsign1",
>>>
>>> +              "affiliateCallSign": "callsign2"
>>>
>>> +            }
>>>
>>> +        )
>>>
>>> +
>>>
>>> +        return response
>>>
>>>
>>> According the the Alexa API:
>>>
>>> You must send an Response event if a request to change or skip channels
>>> was successful. The response should include a context object that reports
>>> the channel after the request completes.
>>>
>>> —
>>> You are receiving this because you authored the thread.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/home-assistant/home-assistant/pull/27605?email_source=notifications&email_token=ABW4E7MX7SMMK2CQCRQXH5DQONYRJA5CNFSM4JAH6NTKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCHY5SMY#pullrequestreview-301062451>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/ABW4E7IC2BOBD2HYGSCW343QONYRJANCNFSM4JAH6NTA>
>>> .
>>>
>>

ochlocracy commented on homeassistant/components/alexa/entities.py:
```diff
@@ -50,7 +51,7 @@
     AlexaStepSpeaker,
     AlexaTemperatureSensor,
-    AlexaThermostatController,
+    AlexaThermostatController
```
```suggestion
    AlexaThermostatController,
```
Black formatting requires commons after last items.

ochlocracy commented:
@Dilbert66 
> What do you think of also implementing the step portion in the skip channel. For now, it treats it as 1. Adding a for loop wil allow the user to skip however many channels they require, just like the StepSpeaker function. Easy to do.

I'm not versed enough in the media_player component to know how to best handle that. But, if the user asks to channel up (skip) 5 channels the handler should make that request to the media_player service, not just the single next as you have it. 

Regarding everything else, I've been working on adding other Alexa Controllers, so just trying to help you out with the simple stuff before the HA reviewers get to it.

javicalle committed:
Move imports in rflink component (#27367)

* Move imports in rflink component

* import order

* import order

* Update __init__.py

* Update __init__.py

I don't understand why tests are failing...

* Fix RFLink imports

* Fix monkeypatch for 'create_rflink_connection'

* isort for rflink classes

ochlocracy commented on homeassistant/components/alexa/entities.py:
```diff
@@ -404,6 +405,9 @@ def interfaces(self):
 
+        if supported & media_player.const.SUPPORT_PLAY_MEDIA:
+            yield AlexaChannelController(self.entity)
+
```
Out of curiosity, what integration are you using/testing this with? 

Unfortunately, it looks like there need to be some additions to the media_player entity model to determine support for channel equipped media players, otherwise there is a potential to expose this controller to all media_player entities that support `SUPPORT_PLAY_MEDIA`.

Michsior14 committed:
move imports in tibber component (#27584)

scop committed:
Upgrade to flake8-docstrings 1.5.0, pytest 5.2.1, and pytest-cov 2.8.1 (#27588)

https://gitlab.com/pycqa/flake8-docstrings/blob/1.5.0/HISTORY.rst
https://docs.pytest.org/en/latest/changelog.html#pytest-5-2-1-2019-10-06
https://pytest-cov.readthedocs.io/en/latest/changelog.html#id1

fabaff committed:
Upgrade Mastodon.py to 1.5.0 (#27598)

Michsior14 committed:
Move imports in syslog (#27602)

Mofeywalker committed:
add content for services.yaml in component media_extractor (#27608)

* add content for services.yaml for media_extractor component

* remove example data

* add empty line on end of file

* Update services.yaml

Michsior14 committed:
move imports in ted5000 component (#27601)

Michsior14 committed:
move imports in tellstick component (#27600)

ggravlingen committed:
Move top level imports (#27597)

Santobert committed:
Improve neato tests (#27578)

* Improve tests

* Rename account to configflow

* configflow to config_flow

* Patch pybotvac instead of own code

ggravlingen committed:
Refactor Tradfri base class (#27589)

* Refactor Tradfri base class

* Clarify doc

* Fix pylint

* Review fix

* Move

tulindo commented:
@Dilbert66 and @ochlocracy sorry disturbing you. 
Few days ago I made a pull request that implements che ChannelController interface that has not yet been reviewed.
I choosed to define 3 new services at media player level (next/prev_channel and channel_set).
Regards, 
  Paolo
PS: @ochlocracy... Here you are 2 different approach to the same problem... I think it's up to you to choose one of the two.

balloob committed:
Google Assistant Local SDK (#27428)

* Local Google

* Fix test

* Fix tests

danielperna84 committed:
Update pyhomematic to 0.1.61 (#27620)

Dilbert66 commented:
Hi Paolo I'll have a look at your pull request. Perhaps it will meet my
needs as well. I don't mind removing my request as long as we end up with a
good versatile solution

On Sun., Oct. 13, 2019, 5:02 p.m. Paolo Tuninetto <notifications@github.com>
wrote:

> @Dilbert66 <https://github.com/Dilbert66> and @ochlocracy
> <https://github.com/ochlocracy> sorry disturbing you.
> Few days ago I made a pull request that implements che ChannelController
> interface that has not yet been reviewed.
> I choosed to define 3 new services at media player level
> (next/prev_channel and channel_set).
> Regards,
> Paolo
> PS: @ochlocracy <https://github.com/ochlocracy>... Here you are 2
> different approach to the same problem... I think it's up to you to choose
> one of the two.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/27605?email_source=notifications&email_token=ABW4E7OYOXOGILLSR37ZUQ3QOOEGTA5CNFSM4JAH6NTKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEBDAIDQ#issuecomment-541459470>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABW4E7MJ67ONP5GFG2GK7SLQOOEGTANCNFSM4JAH6NTA>
> .
>

homeassistant committed:
[ci skip] Translation update

Dilbert66 commented:
Hi Paolo, I had a quick look at your implementation and it looks good.  The
main difference with your changes are that they require other changes to
the core media player, and my implementation is using the current available
function play_media for channel change since it does have a "channel"
option.  The skip channel is done via nexttrack which I agree is not
technically a channel function but depending on how the media_player is
configured will work fine.   Your approach is perhaps the more "correct"
way as it adds TV specific changes to the media player. Mine is a bit less
intrusive.   This is really a decision by the main developers as to how
they want to enhance the media player's versatility.   If they want to add
the extra features to the media player, I'd be happy with that as well.  I
still need to correct some changes on mine so I will wait till yours is
fully reviewed before completing the pull request on my end.     If I may
offer a suggestion , you might also want to enhance the skipchannel
function to allow multiple channel skip's since that capability is offered
in the alexa API. I don't have that now but intend on adding it.

Regards, Alain

On Sun, Oct 13, 2019 at 5:02 PM Paolo Tuninetto <notifications@github.com>
wrote:

> @Dilbert66 <https://github.com/Dilbert66> and @ochlocracy
> <https://github.com/ochlocracy> sorry disturbing you.
> Few days ago I made a pull request that implements che ChannelController
> interface that has not yet been reviewed.
> I choosed to define 3 new services at media player level
> (next/prev_channel and channel_set).
> Regards,
> Paolo
> PS: @ochlocracy <https://github.com/ochlocracy>... Here you are 2
> different approach to the same problem... I think it's up to you to choose
> one of the two.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/27605?email_source=notifications&email_token=ABW4E7OYOXOGILLSR37ZUQ3QOOEGTA5CNFSM4JAH6NTKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEBDAIDQ#issuecomment-541459470>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABW4E7MJ67ONP5GFG2GK7SLQOOEGTANCNFSM4JAH6NTA>
> .
>

Dilbert66 commented:
Currently i'm using a custom media_player called (
https://community.home-assistant.io/t/smartir-control-your-climate-tv-and-fan-devices-via-ir-rf-controllers/
<https://community.home-assistant.io/t/smartir-control-your-climate-tv-and-fan-devices-via-ir-rf-controllers/100798/526>)
SmartIR  that uses a broadlink IR device to control a tv cable box.  It
basically uses all standard media_player functions.  The channel change
piece is extremely handy to control my tv as it handles all volume,
channel, play/pause (PVR),etc.
From what I can see, the media_play function being available for channel
changes should not be an issue as long as the media_player handles the
function correctly and validates the media type field being sent before
processing the request.  The type being used is 'channel' and is part of
the allowed data types.  I can't see any other way without adding new
SUPPORT_xxx flags to specify tv functions only I was trying to avoid doing
too many changes to the media_player model.  I suspect the media_player
entity model is due for some new capabilities enhancements.



On Sun, Oct 13, 2019 at 4:25 PM ochlocracy <notifications@github.com> wrote:

> *@ochlocracy* commented on this pull request.
> ------------------------------
>
> In homeassistant/components/alexa/entities.py
> <https://github.com/home-assistant/home-assistant/pull/27605#discussion_r334295090>
> :
>
> > +        if supported & media_player.const.SUPPORT_PLAY_MEDIA:
> +            yield AlexaChannelController(self.entity)
> +
>
> Out of curiosity, what integration are you using/testing this with?
>
> Unfortunately, it looks like there need to be some additions to the
> media_player entity model to determine support for channel equipped media
> players, otherwise there is a potential to expose this controller to all
> media_player entities that support SUPPORT_PLAY_MEDIA.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/27605?email_source=notifications&email_token=ABW4E7P7VXJY7YLLCDIUHLDQON737A5CNFSM4JAH6NTKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCHY6ITQ#pullrequestreview-301065294>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABW4E7M6CYJMA34AXMPXAZLQON737ANCNFSM4JAH6NTA>
> .
>

Dilbert66 committed:
Added response context to payload directive for channelcontrol
Added check for uri,affiliateCallSign

Dilbert66 committed:
Added response context to payload directive for channelcontrol
Added check for uri,affiliateCallSign

Dilbert66 committed:
Formatting fixes

Dilbert66 committed:
Formatting fixes

Dilbert66 committed:
Fix code for black validation

Dilbert66 committed:
Fix code for black validation

Dilbert66 committed:
More formatting fixes

Dilbert66 committed:
More formatting fixes

Dilbert66 committed:
Corrected typo, other format fixes

Dilbert66 committed:
Corrected typo, other format fixes

Dilbert66 committed:
Formating fun....

Dilbert66 committed:
Formating fun....

Dilbert66 committed:
Formatting...

Dilbert66 committed:
Formatting...

jjlawren committed:
Rewrite Plex tests (#27624)

Michsior14 committed:
fix: exception after kaiterra api call timeout (#27622)

stevendlander committed:
Issue #27288 Moving imports to top for tesla component (#27618)

Mofeywalker committed:
add content for services.yaml for ccomponent stream (#27610)

tulindo commented:
Hi Alain, as I said, we simply had different  approach to the same problem. Both with pros and cons. 
Let's wait until they choose

rishatik92 committed:
Update yandex transport after api change (#27591)

* yandex maps api changed ("threads" in "Transport" added), ya_ma=>0.3.8 bug_fixed

* Update homeassistant/components/yandex_transport/sensor.py

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

* additional fix

* reformat

* fix mistake

stevendlander committed:
Issue #27288 Move imports to top for FFMPEG (#27613)

exxamalte committed:
move imports to top-level (#27630)

pvizeli committed:
Update azure-pipelines-wheels.yml for Azure Pipelines

Dilbert66 committed:
Single quote to double. Variable name change

Dilbert66 committed:
Single quote to double. Variable name change

Dilbert66 committed:
fix introduced whitespace to fix lint check

Dilbert66 committed:
fix introduced whitespace to fix lint check

Michsior14 committed:
move imports in synology_srm component (#27603)

Bouni committed:
moved imports to top level (#27632)

Bouni committed:
moved imports to top level (#27634)

exxamalte committed:
Move imports in darksky component (#27633)

* move imports to top-level

* modify patch path

* removed unused mocks and patches

crazyfx1 committed:
Fix temperature and heating mode (#27604)

Bouni committed:
moved imports to top level (#27640)

OttoWinter committed:
ESPHome Fix intermediary state published (#27638)

Fixes https://github.com/esphome/issues/issues/426

Dilbert66 committed:
Removed extra line found by flake8

Dilbert66 committed:
Removed extra line found by flake8

Dilbert66 committed:
Removed extra line found by flake8

homeassistant commented:
Hello @Dilbert66,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`25c81fbd0811689a369d939ff6f42e7e447d6949`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;27605&#x2F;commits&#x2F;25c81fbd0811689a369d939ff6f42e7e447d6949) - This commit has something that looks like an email address (alt@ha.hass99.duckdns.org). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

Dilbert66 committed:
Merge branch 'dev' of https://github.com/Dilbert66/home-assistant into dev

homeassistant commented:
Hello @Dilbert66,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`25c81fbd0811689a369d939ff6f42e7e447d6949`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;27605&#x2F;commits&#x2F;25c81fbd0811689a369d939ff6f42e7e447d6949) - This commit has something that looks like an email address (alt@ha.hass99.duckdns.org). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

Bouni committed:
moved imports to top level (#27641)

delphiki committed:
Fix html5 notification documentation url (#27636)

javicalle committed:
Apply isort on rfxtrx classes (#27615)

* Move imports in rfxtrx component

* Apply isort on rfxtrx files

* Update test_switch.py

OttoWinter committed:
Fix ESPHome climate preset mode refactor (#27637)

Fixes https://github.com/home-assistant/home-assistant/issues/25613

stevendlander committed:
Refactor imports for tensorflow (#27617)

* Refactoring imports for tensorflow

* Removing whitespace spaces on blank line 110

* Moving tensorflow to try/except block

* Fixed black formatting

* Refactoring try/except to if/else

Bouni committed:
Move imports in bluesound component (#27502)

* moved imports to top level

* changed import order

* changed import order

emontnemery committed:
Fix ZHA regressions caused by "Support async validation of device trigger" (#27401)

* Revert "Support async validation of device trigger (#27333)"

This reverts commit fdf4f398a79e775e11dc9fdd391a8b53f7b773c5.

* Revert only ZHA changes

* Fix whitespace

* Restore ZHA changes but add check to make sure ZHA is loaded

* Address review comment

* Remove additional check

Michsior14 committed:
move imports in squeezebox component (#27650)

scop committed:
Typing misc fixes (#27543)

* Make async_get_conditions return type hint more specific

* Exclude script/scaffold/templates/ from pre-commit mypy

bramkragten committed:
Updated frontend to 20191014.0 (#27661)

Dilbert66 committed:
Added ChannelController test. Modified StepSpeaker AdjustVolume test to include test for value volumeStepDefault that also gets sent in the API.  It is not in the online docs but is returned in the payload. Also other black fixes.

Dilbert66 committed:
Added ChannelController test. Modified StepSpeaker AdjustVolume test to include test for value volumeStepDefault that also gets sent in the API.  It is not in the online docs but is returned in the payload. Also other black fixes.

Dilbert66 committed:
Added ChannelController test. Modified StepSpeaker AdjustVolume test to include test for value volumeStepDefault that also gets sent in the API.  It is not in the online docs but is returned in the payload. Also other black fixes.

homeassistant commented:
Hello @Dilbert66,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`25c81fbd0811689a369d939ff6f42e7e447d6949`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;27605&#x2F;commits&#x2F;25c81fbd0811689a369d939ff6f42e7e447d6949) - This commit has something that looks like an email address (alt@ha.hass99.duckdns.org). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

javicalle committed:
Move imports in panasonic_viera component (#27665)

Kane610 committed:
Improve discovery title (#27664)

javicalle committed:
Move imports in panasonic_bluray component (#27658)

ochlocracy committed:
Update Unlock directive for Alexa LockController (#27653)

* Update the Alexa.LockController Unlock directive to include the lockState property in the context of the response.

* Added Test for Alexa.LockController Unlock directive to include the lockState property in the context of the response.

Michsior14 committed:
move imports in statsd component (#27649)

Michsior14 committed:
Move imports in steam_online component (#27648)

* move imports in steam_online component

* fix: import reassigment

Michsior14 committed:
move imports in stream component (#27647)

Michsior14 committed:
Move imports in switchmate component (#27646)

* move imports in switchmate component

* fix: bring back pylint ignore line

Dilbert66 committed:
Merge branch 'dev' of https://github.com/Dilbert66/home-assistant into dev

homeassistant commented:
Hello @Dilbert66,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`25c81fbd0811689a369d939ff6f42e7e447d6949`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;27605&#x2F;commits&#x2F;25c81fbd0811689a369d939ff6f42e7e447d6949) - This commit has something that looks like an email address (alt@ha.hass99.duckdns.org). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

balloob committed:
Remove direct authentication via trusted networks or API password (#27656)

* Remove direct authentication via trusted networks and API password

* Fix tests

Dilbert66 committed:
Removed unecessary test line from code

Dilbert66 committed:
Removed unecessary test line from code

Dilbert66 committed:
Removed unecessary test line from code

homeassistant commented:
Hello @Dilbert66,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`25c81fbd0811689a369d939ff6f42e7e447d6949`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;27605&#x2F;commits&#x2F;25c81fbd0811689a369d939ff6f42e7e447d6949) - This commit has something that looks like an email address (alt@ha.hass99.duckdns.org). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

Dilbert66 committed:
Merge branch 'dev' of https://github.com/Dilbert66/home-assistant into dev

homeassistant commented:
Hello @Dilbert66,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`25c81fbd0811689a369d939ff6f42e7e447d6949`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;27605&#x2F;commits&#x2F;25c81fbd0811689a369d939ff6f42e7e447d6949) - This commit has something that looks like an email address (alt@ha.hass99.duckdns.org). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

Dilbert66 committed:
Test push to correct cla error.

Dilbert66 committed:
Test push to correct cla error.

homeassistant commented:
Hello @Dilbert66,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`25c81fbd0811689a369d939ff6f42e7e447d6949`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;27605&#x2F;commits&#x2F;25c81fbd0811689a369d939ff6f42e7e447d6949) - This commit has something that looks like an email address (alt@ha.hass99.duckdns.org). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

caronc committed:
Add Apprise notification integration (#26868)

* Added apprise notification component

* flake-8 fixes; black formatting + import merged to 1 line

* pylint issues resolved

* added github name to manifest.json

* import moved to top as per code review request

* manifest formatting to avoid failing ci

* .coveragerc updated to include apprise

* removed block for written tests

* more test coverage

* formatting as per code review

* tests converted to async style as per code review

* increased coverage

* bumped version of apprise to 0.8.1

* test that mocked entries are called

* added tests for hass.service loading

* support tags for those who identify the TARGET option

* renamed variable as per code review

* 'assert not' used instead of 'is False'

* added period (in case linter isn't happy)

Dilbert66 committed:
Merge branch 'dev' of https://github.com/Dilbert66/home-assistant into dev

Dilbert66 commented:
I'm closing this pull request as i need to rebuild my forked copy after having issues with my devevlopment environment.  Will re pull once all rectified.
